src/docfx/build/xref/XrefMap.cs

```diff
@@ -156,20 +153,49 @@ private static XrefSpec LoadSchemaDocument(List<Error> errors, JToken token, Doc
 
                 var (schemaErrors, content) = JsonUtility.ToObject(token, schema.Type, transform: TransformContent);
                 errors.AddRange(schemaErrors);
-                return xref;
+                return xrefs.Values;
 
-                object TransformContent(DataTypeAttribute attribute, object value)
+                object TransformContent(DataTypeAttribute attribute, object value, string jsonPath, string fieldName)
                 {
                     if (attribute is UidAttribute)
                     {
-                        if (!string.IsNullOrEmpty((string)value))
+                        var level = GetLevelFromPath(jsonPath);
+                        if (!xrefs.ContainsKey(level))
+                        {
+                            xrefs.Add(level, new XrefSpec { Uid = (string)value, Href = file.SitePath });
+                        }
+                        else
+                        {
+                            errors.Add(Errors.MultipleUid());
```

Comment:
>Errors.MultipleUid() [](start = 39, length = 20)

Move this check when load schema? It seems like a schema error, which defines 2 UidAttribute to a single object.

Reply:
Would be hard to do it inside schema validation, there we only validate the current property value, this is kind of related to business logic, i.e.: multiple uids are not allowed

Reply:
And actually, this is part of schema validation since TransformContent is passed into JsonUtility.ToObject. It is just used to isolate xref logic in XrefMap.cs, just like we did in BuildPage.cs, to isolate SDP logic there.

Reply:
do we still want to have this design, i.e. UidAttribute must be named as "uid"?

---
In reply to: [219438486](https://github.com/dotnet/docfx/pull/3443#discussion_r219438486) [](ancestors = 219438486)

Reply:
I think we can remove this restriction

Reply:
Personally prefer to preserve this to make it as a convension

---
In reply to: [220049680](https://github.com/dotnet/docfx/pull/3443#discussion_r220049680) [](ancestors = 220049680,219438486)